Source/JavaScriptCore/wasm/WasmSectionParser.cpp

```diff
@@ -719,6 +719,36 @@ auto SectionParser::parseFunctionType(uint32_t position, RefPtr<TypeDefinition>&
     return { };
 }
 
+auto SectionParser::parsePackedType(PackedType& packedType) -> PartialResult
+{
+    int8_t kind;
+    WASM_PARSER_FAIL_IF(!parseInt7(kind), "invalid type in struct field or array element");
+    if (isValidPackedTypeKind(kind)) {
+        packedType = static_cast<PackedType>(kind);
+        return { };
+    }
+    return fail("expected a packed type but got ", kind);
+}
+
+auto SectionParser::parseStorageType(StorageType& storageType) -> PartialResult
+{
+    ASSERT(Options::useWebAssemblyGC());
+
+    int8_t kind;
+    WASM_PARSER_FAIL_IF(!peekInt7(kind), "invalid type in struct field or array element");
+    if (isValidTypeKind(kind)) {
+        Type elementType;
+        WASM_PARSER_FAIL_IF(!parseValueType(m_info, elementType), "invalid type in struct field or array element");
+        storageType = StorageType(std::as_const(elementType));
```

Comment:
Is `std::as_const` necessary here? Also I believe `{}`-initialization (https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-list) is generally preferred in JSC for structs like this, something like `storageType = StorageType { ... }` (see uses of `Type { ... }` elsewhere in the file too).